### PR TITLE
fix: Changes in function process_dataargs to support the current implementation

### PR DESCRIPTION
**Pull request by Abhishek-TAMU**
### Description of the change

1- Adding unit test case for testing function `process_dataargs` with current way of tuning.
2- Changes to make the case of dataset with `input, output` key work with current way of tuning and make `test_process_dataargs_pretokenized` handler work.

### Related issue number

https://github.ibm.com/ai-foundation/watson-fm-stack-tracker/issues/1428

### How to verify the PR

Run current way of tuning for multiple cases:
- Case with using dataset with `text and label` as a single sequence, using [this dataset](https://github.com/Abhishek-TAMU/fms-hf-tuning/blob/data_preprocessor_dushyant/tests/data/twitter_complaints_small.jsonl)

Command used:
```
python tuning/sft_trainer.py  \
--model_name_or_path Maykeye/TinyLLama-v0  \
--training_data_path tests/data/twitter_complaints_small.jsonl  \
--output_dir outputs/full-tuning  \
--num_train_epochs 5  \
--per_device_train_batch_size 2  \
--gradient_accumulation_steps 1  \
--learning_rate 1e-5  \
--response_template "\n### Label:"  \
--dataset_text_field "output" \
--use_flash_attn false \
--torch_dtype "float32"
```

- Case with `pre-tokenized dataset` using [this dataset](https://github.com/Abhishek-TAMU/fms-hf-tuning/blob/data_preprocessor_dushyant/tests/data/twitter_complaints_tokenized_with_maykeye_tinyllama_v0.jsonl).

Command used:
```
python tuning/sft_trainer.py  \
--model_name_or_path Maykeye/TinyLLama-v0  \
--training_data_path tests/data/twitter_complaints_tokenized_with_maykeye_tinyllama_v0.jsonl \
--output_dir outputs/full-tuning  \
--num_train_epochs 5  \
--per_device_train_batch_size 2  \
--gradient_accumulation_steps 1  \
--learning_rate 1e-5  \
--use_flash_attn false \
--torch_dtype "float32"
```

- Case with `input, output` key in dataset using [this dataset](https://github.com/Abhishek-TAMU/fms-hf-tuning/blob/data_preprocessor_dushyant/tests/data/twitter_complaints_input_output.jsonl)

Command used:
```
python tuning/sft_trainer.py  \
--model_name_or_path Maykeye/TinyLLama-v0  \
--training_data_path tests/data/twitter_complaints_input_output.jsonl  \
--output_dir outputs/full-tuning  \
--num_train_epochs 5  \
--per_device_train_batch_size 2  \
--gradient_accumulation_steps 1  \
--learning_rate 1e-5  \
--use_flash_attn false \
--torch_dtype "float32"
```

### Was the PR tested

- [ ] I have added >=1 unit test(s) for every new method I have added.
- [x] I have ensured all unit tests pass

### CODE REVIEW

File: tuning/data/data_handlers.py [hunk 1]
```diff
@@ -38,8 +38,11 @@ def tokenize_and_apply_input_masking(
     # TODO: Eventually move the code here
     combined = combine_sequence(input, output, eos_token=tokenizer.eos_token)
 
-    tokenized_comb_seqs = tokenizer(combined, **tokenizer_kwargs)
-    tokenized_input = tokenizer(input, **tokenizer_kwargs)
+    fn_kwargs = tokenizer_kwargs.get("fn_kwargs", {})
+    tokenizer_inner_kwargs = fn_kwargs.get("tokenizer_kwargs", {})
+
+    tokenized_comb_seqs = tokenizer(combined, **tokenizer_inner_kwargs)
+    tokenized_input = tokenizer(input, **tokenizer_inner_kwargs)
```
Comment (Abhishek-TAMU) [inline tuning/data/data_handlers.py:45]: Reference of discussion for this change: https://github.com/foundation-model-stack/fms-hf-tuning/pull/381#discussion_r1851174125

File: tuning/data/setup_dataprocessor.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def process_dataargs(
         kwargs = {
             "fn_kwargs": fn_kwargs,
             "batched": False,
-            "remove_columns": [JSON_INPUT_KEY, JSON_OUTPUT_KEY],
+            "remove_columns": "all",
```
Comment (Abhishek-TAMU) [inline tuning/data/setup_dataprocessor.py:121]: In current implementation, dataset just have these [columns](https://github.com/dushyantbehl/fms-hf-tuning/blob/163fe34fca481ae06549a73d2785f8dde4d39c88/tuning/data/data_handlers.py#L52) `input_ids,  labels", attention_mask`, hence every other columns in the dataset needed to be removed.

File: tuning/data/data_handlers.py (hunk 1 above)
Comment (dushyantbehl) [reply tuning/data/data_handlers.py:45]: Thanks @Abhishek-TAMU
